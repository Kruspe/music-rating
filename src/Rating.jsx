import React, { useState } from 'react';
import { Rating as RatingMaterialUI } from '@material-ui/lab';

const Rating = () => {
  const [rating, setRating] = useState(0);
  return (
    <RatingMaterialUI value={rating} onChange={(event, value) => setRating(value)} />
  );
};

export default Rating;
