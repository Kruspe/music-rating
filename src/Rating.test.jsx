import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Rating as RatingMaterialUI } from '@material-ui/lab';
import Rating from './Rating';

let wrapper;
describe('Rating', () => {
  beforeEach(() => {
    wrapper = shallow(<Rating />);
  });
  it('should render correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  describe('bandName', () => {
    it('should have correct initial value', () => {
      expect(wrapper.find('#bandName').prop('value')).toBe('');
    });
    it('should update TextField onChange', () => {
      wrapper.find('#bandName').prop('onChange')({ target: { value: 'bandName' } });
      expect(wrapper.find('#bandName').prop('value')).toBe('bandName');
    });
  });

  describe('rating', () => {
    it('should have correct initial state', () => {
      expect(wrapper.find(RatingMaterialUI).prop('value')).toBe(0);
    });
    it('should update rating onChange', () => {
      wrapper.find(RatingMaterialUI).prop('onChange')({}, 2);
      expect(wrapper.find(RatingMaterialUI).prop('value')).toBe(2);
    });
  });

  describe('comment', () => {
    it('should have correct initial state', () => {
      expect(wrapper.find('#comment').prop('value')).toBe('');
    });
    it('should update rating onChange', () => {
      wrapper.find('#comment').prop('onChange')({ target: { value: 'comment' } });
      expect(wrapper.find('#comment').prop('value')).toBe('comment');
    });
  });
});
