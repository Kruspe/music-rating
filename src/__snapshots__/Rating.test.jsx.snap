// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <WithStyles(ForwardRef(TextField))
      id="bandName"
      label="Band Name"
      onChange={[Function]}
      value=""
      variant="outlined"
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <WithStyles(ForwardRef(Rating))
      onChange={[Function]}
      value={0}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={5}
  >
    <WithStyles(ForwardRef(TextField))
      fullWidth={true}
      id="comment"
      label="Comment"
      onChange={[Function]}
      value=""
      variant="outlined"
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
